Sources/DataPipeline/CustomerIO.swift

```diff
@@ -28,6 +29,12 @@ public extension CustomerIO {
         CustomerIO.shared.setDebugLogsEnabled(logLevel == .debug)
 
         initialize(implementation: implementation, diGraph: newDiGraph)
+
+        // Handle logged-in user from Journeys to CDP and check
+        // if any unprocessed tasks are pending in the background queue.
+        let migrationHandler = DataPipelineMigrationHandler(implementation: implementation)
+        let migrationAssistant = DataPipelineMigrationAssistant(handler: migrationHandler, diGraph: newDiGraph)
+        migrationAssistant.performMigration(for: DataPipeline.shared.analytics.userId)
```

Comment:
[Migration will happen only if the userId is `nil`](https://github.com/customerio/customerio-ios/blob/bb3ed4a673b8a5dd0e01244c2464ca6d4c026173/Sources/Migration/DataPipelineMigrationAssistant.swift#L34). This check and handling is done by MigrationAssistant.

Reply:
I'm okay if the rest of the team agrees. However, looking at the code, the `for` parameter looks like expecting an identifier for the migration target, but the implementation seems to operate inversely, which I find quite confusing.

Reply:
I do agree with Rehan that I would expect the `for` parameter to be an identifier to migrate from, not migrate to. Making the code harder to read but also prone to miss-use. 

Luckily, the code Aman is talking about has automated tests written around it so that makes me less concerned about introducing a bug in the future if there is miss-use.

Reply:
[We do not need the identifier now.](https://github.com/customerio/customerio-ios/blob/d21018d192d2f91dc9f352bcdde0fcb347777b1e/Sources/DataPipeline/CustomerIO.swift#L32) This conversation is outdated hence resolving it.